src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,48 @@
 import React from 'react';
+import { ExtendedPost } from '../../types/Post';
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo';
+import './PostInfo.scss';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+type Props = {
+  post: ExtendedPost
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const {
+    title,
+    body,
+    user,
+    comments,
+  } = post;
+
+  return (
+    <>
+      <div className="PostInfo">
+        <div className="PostInfo__header">
+          <h3 className="PostInfo__title">{title}</h3>
+
+          {(user) && (
+            <p>
+              {' Posted by  '}
+              <UserInfo user={user} />
+            </p>
+          )}
+
+        </div>
+
+        <p className="PostInfo__body">
+          {body}
+        </p>
+
+        {!post.comments.length
+          ? (
+            <b data-cy="NoCommentsMessage">No comments yet</b>
+          )
+          : (
+            <CommentList comments={comments} />
+          )}
```

Comment:
is there a reason to start from the negative? I mean it would be more logical to start with the condition `post.comments.length` because it is the main logic